### PR TITLE
Spontanaeous reboot hardening documentation

### PR DESCRIPTION
**Pull request by oleorhagen**
Changelog: None

Signed-off-by: Ole Petter <ole.orhagen@cfengine.com>

### CODE REVIEW

**Comment by oleorhagen** [comment]
Do not merge until the client functionality is merged as well. Ref: MEN-1187

**Comment by drewmoseley** [comment]
@oleorhagen in addition to the wording comments, it's unclear to me what needs to happen next.  Does the user need to do anything specific in the mentioned state scripts?

**Comment by oleorhagen** [comment]
@drewmoseley I think u forgot the comments. And also, have a look at the last commit now ;)

**Comment by drewmoseley** [comment]
@oleorhagen I only see the original commit still.  Did you push something new?

**Comment by oleorhagen** [comment]
@drewmoseley now I have :)

File: 04.Artifacts/07.State-scripts/docs.md [hunk 1]
```diff
@@ -55,6 +55,12 @@ If a script returns `0` Mender proceeds, but if it returns `1` the update is abo
 In addition, return code `21` is used for the [retry-later](#retry-later) feature.
 All other return codes are reserved for future use by Mender and should not be used.
 
+## Powerloss hardening
+If the system loses power during an update, there are two cases that will need to be handled. The mender client will execute state scripts as follows:
+- The new partition is not yet committed, in which case the client guarantees that the `ArtifactFailure` scripts will be rerun on the original partition, before resuming normal execution.
+- Power is lost during the execution of `ArtifactCommit_Leave`, in which case the `ArtifactCommit_Leave` scripts will be rerun on startup before resuming normal execution.
```
Comment (drewmoseley) [inline 04.Artifacts/07.State-scripts/docs.md]: For extra clarity, I think we should state something like "rerun, on the newly installed partition, at startup before resuming normal execution with the successfully installed update.

**Pull request merged**